src/main/java/org/rulez/demokracia/pdengine/VoteRegistry.java

```diff
@@ -19,11 +19,15 @@ public String obtainBallot(String id, String adminKey) {
 
 		if (adminKey.equals(vote.adminKey))
 			vote.increaseRecordedBallots("admin");
-		else if (adminKey.equals("user"))
+
+		else if (adminKey.equals("user")) {
 			if (!userHasAllAssurance(vote.neededAssurances))
 				throw new IllegalArgumentException("The user does not have all of the needed assurances.");
-			else
-				vote.increaseRecordedBallots(getWsUserName());
+			if (! vote.ballots.isEmpty())
```

Comment:
This checks whether there are any ballots on the vote. Should be checked whether the given user have a ballot. Most probably it should be done in increaseRecordedBallots.
If your tests records a ballot with the adminKey, then two "user" ballots, You will get this straight.

Reply:
OK. Now the concept is clear. Anon admin / Simpleuser / Admin="user" all mean that the user is a simple voter, identified by the session. I would have been helpful to know that before.

Reply:
We are all learning this kind of development.